src/main/java/com/hms_networks/sc/cumulocity/data/CConnectorDataMgr.java

```diff
@@ -0,0 +1,506 @@
+            }
+            final boolean startNewTimeTracker = true;
+            datapointsReadFromQueue =
+                HistoricalDataQueueManager.getFifoNextSpanDataAllGroups(startNewTimeTracker);
+          }
+
+          // Reset queue force reset tag value to false, if true
+          if (queueDiagnosticForceResetTag != null && forceReset) {
+            queueDiagnosticForceResetTag.setTagValueAsInt(
+                QUEUE_DIAGNOSTIC_TAG_FORCE_RESET_FALSE_VALUE);
+          }
+
+          Logger.LOG_DEBUG(
+              "Read " + datapointsReadFromQueue.size() + " data points from the historical log.");
+
+          // Reset failure counter
+          queuePollFailCount = 0;
+
+          // Check if queue is behind
+          try {
+            long queueBehindMillis = HistoricalDataQueueManager.getQueueTimeBehindMillis();
+            if (queueBehindMillis >= QUEUE_DATA_POLL_BEHIND_MILLIS_WARN) {
+              Logger.LOG_WARN(
+                  "The historical data queue is running behind by "
+                      + SCTimeUtils.getDayHourMinSecsForMillis(
+                          (int) queueBehindMillis, "days", "hours", "minutes", "seconds"));
+            } else {
+              // Queue not running behind, reset to 0 for updating debug tag
+              queueBehindMillis = 0;
+            }
+
+            // Update queue debug tag
+            if (queueDiagnosticRunningBehindSecondsTag != null) {
+              queueDiagnosticRunningBehindSecondsTag.setTagValueAsLong(
+                  SCTimeUnit.MILLISECONDS.toSeconds(queueBehindMillis));
+            }
+          } catch (IOException e) {
+            Logger.LOG_SERIOUS("Unable to detect if historical data queue is running behind.");
+            Logger.LOG_EXCEPTION(e);
+          }
+
+          // Send data via MQTT
+          if (datapointsReadFromQueue.size() > 0) {
+            Map childDeviceMessageMap = new HashMap();
+            for (int i = 0; i < datapointsReadFromQueue.size(); i++) {
+              DataPoint datapoint = (DataPoint) datapointsReadFromQueue.get(i);
+
+              // Get and split tag name of data point
+              String datapointTagName = datapoint.getTagName();
+              String[] datapointSplitTagName = getSplitTagName(datapointTagName);
+
+              // Get data point information
+              String childDevice = datapointSplitTagName[0];
+              String fragment = "\"" + datapointSplitTagName[1] + "\"";
+              String series = datapointSplitTagName[2];
+              String value = datapoint.getValueString();
+              String unit = datapoint.getTagUnit();
+              String time = SCTimeUtils.getIso8601FormattedTimestampForDataPoint(datapoint);
+
+              // Build payload string contents
+              String payloadString;
+              if (datapoint.getType() == DataType.STRING) {
+                // Handle strings as a basic event (use filler if no value/only quotes)
```

Comment:
No value and only quotes are not string literals, so I'm not sure why quotes would be needed.

Reply:
it reads as value or only instead of no value or only quotes at a fast skim through. Some sort of separation out of the rest of the comment would improve the readability just a little